bot/cogs/reminders.py

```diff
@@ -275,9 +286,12 @@ def __init__(self, bot: Bot):
             json={'content': content}
         )
 
+        # Parse the reminder expiration back into a datetime for the confirmation message
+        expiration = datetime.fromisoformat(reminder['expiration'][:-1])
```

Comment:
We've been using dateutil for parsing cause I suppose it's less fragile:

https://github.com/python-discord/bot/blob/master/bot/utils/time.py#L133 (ignore the microseconds replacement)

Looks like this cog has several places that could be using this instead.